### ISSUE TITLE
Startup option gets reset

### ISSUE THREAD

**Issue by NYR94** [vanowm/MasterPasswordPlus#134]
If 'EXIT' is chosen for the Startup option if cancel is clicked, it works once then gets reset back to 'CANCEL'. Is there any way to fix it so once 'EXIT' is chosen it stays ?

Thanks

**Comment by vanowm** [comment]
I can't reproduce it on my end

What browser and version?

**Comment by NYR94** [comment]
Firefox developer Edition 58

**Comment by vanowm** [comment]
Sorry, but this extension doesn't work in FF57+ and never will.

**Comment by NYR94** [comment]
It works when I enable 'extensions.legacy.enabled;true' in about:config.
The add-on works, only problem is it resets back to 'Do Nothing' when set to 'EXIT' in the Startup options.

**Comment by vanowm** [comment]
It doesn't work at all on my computer. Just look in browser console, it chokes on startup and none of the settings are present.
If you see any non-bold settings in `about:config?filter=extensions.masterpasswordplus` it means no default preferences exist.

**Comment by NYR94** [comment]
It works on 2 computers of mine. All of the 'masterpasswordplus' preferences in about:config are bold.
When the option is set to 'Exit', extensions.masterPasswordPlus.startupfail; goes from 0 to 1 then gets reset back to 0 after a login.

**Comment by vanowm** [comment]
That's because MP+ does not initialize properly due to Ci.nsISupportsString no longer exists in Firefox (and probably other issues).
Try this, right click on extensions.masterPasswordPlus.startupfail and select reset, it should reset to an integer, not string.
Restart browser and see if that preference still exists in about:config

**Comment by NYR94** [comment]
When I reset that integer, it does become a string, but the option gets reset back to 'Do Nothing'.
When I change it back to 'Exit' it becomes an integer again and the same thing happens again.
I need to password protect my browser and this was the only add-on that I found. Everything worked fine until I noticed once I hit cancel at the password prompt it still let me in the browser, even though I set it to 'Exit'.
